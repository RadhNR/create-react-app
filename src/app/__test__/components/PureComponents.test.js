import React from 'react';
import { shallow, render } from 'enzyme';
import PureComponents from '../../components/PureComponent';

import renderer from 'react-test-renderer';

describe('When PureComponent is given', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = renderer.create(
      <PureComponents />
    );
  });
  it('should render the component', () => {
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('When changeText() is called', () => {
    beforeEach(() => {
      wrapper.getInstance().changeText();
    });

    it('should display updated value', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('When button is clicked', () => {
    beforeEach(() => {
      wrapper.toJSON().children[1].props.onClick();
    });

    it('should display updated value', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });  
});

// describe('When PureComponent is given', () => {
//   let wrapper;
//   beforeEach(() => {
//     wrapper = shallow(<PureComponents />);
//   });

//   it('render snapshot', () => {
//     expect(wrapper).toMatchSnapshot();
//   });

//   // it('renders without crashing', () => {
//   //   expect(wrapper).toHaveLength(1);
//   // });

//   // it('should have div with class', () => {
//   //   expect(wrapper.find('.pureComp')).toHaveLength(1);
//   // });

//   it('should have text displayed', () => {
//     //expect(wrapper.find('.data').text()).toEqual('PureComponent Text : ' + wrapper.state().age)
//     expect(wrapper.find('.data').text()).toMatchSnapshot();
//   });

//   it('should have updated text displayed', () => {
//     wrapper.find('button').simulate('click');
//     expect(wrapper.find('.data').text()).toEqual('PureComponent Text : ' + wrapper.state().age)
//     //expect(wrapper.find('.data').text()).toMatchSnapshot();
//   });

// });

