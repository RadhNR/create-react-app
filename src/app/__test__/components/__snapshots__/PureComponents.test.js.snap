// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When PureComponent is given When button is clicked should display updated value 1`] = `
<div
  className="pureComp1"
>
  <div
    className="data"
  >
    PureComponent Text : 
    705
  </div>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;

exports[`When PureComponent is given When changeText() is called should display updated value 1`] = `
<div
  className="pureComp1"
>
  <div
    className="data"
  >
    PureComponent Text : 
    705
  </div>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;

exports[`When PureComponent is given should render the component 1`] = `
<div
  className="pureComp1"
>
  <div
    className="data"
  >
    PureComponent Text : 
    304
  </div>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;
